GoogleCloudExtension/ProjectTemplates/ProjectTemplate.Tests/ComHelper.cs

```
@@ -0,0 +1,86 @@
+    public static class ComHelper
+    {
+        [SuppressMessage("ReSharper", "InconsistentNaming")]
+        private static class HRESULT
+        {
+            public const int S_OK = 0;
+        }
+
+        ///<see href="https://msdn.microsoft.com/en-us/library/windows/desktop/ms678542"/>
+        [DllImport("ole32.dll")]
+        private static extern int CreateBindCtx(uint zero, out IBindCtx ctx);
+
+        /// <summary>
+        /// Get the COM object from the running object table that matches the predicate.
+        /// </summary>
+        /// <param name="predicate">The test to check the monikers against.</param>
+        public static object GetRunningObject(Func<IMoniker, IBindCtx, bool> predicate)
+        {
+            IBindCtx bindCtx;
+            Marshal.ThrowExceptionForHR(CreateBindCtx(0, out bindCtx));
+            IRunningObjectTable rot;
+            bindCtx.GetRunningObjectTable(out rot);
+            var targetMoniker = rot.GetRunningMonikers().FirstOrDefault(m => predicate(m, bindCtx));
+            if (targetMoniker != null)
+            {
+                object dte;
+                Marshal.ThrowExceptionForHR(rot.GetObject(targetMoniker, out dte));
+                return dte;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private static IEnumerable<IMoniker> GetRunningMonikers(this IRunningObjectTable rot)
+        {
+            IEnumMoniker monikersEnum;
+            rot.EnumRunning(out monikersEnum);
+            return monikersEnum.ToEnumerable();
+        }
+
+        private static IEnumerable<IMoniker> ToEnumerable(this IEnumMoniker monikersEnum)
+        {
+            int fetchSize = 1;
+            var monikers = new IMoniker[fetchSize];
+            IntPtr fetchedCount = IntPtr.Zero;
+            while (monikersEnum.Next(fetchSize, monikers, fetchedCount) == HRESULT.S_OK)
```

Should you throw in case of an unexpected error?